api/src/main/java/edu/cornell/mannlib/vitro/webapp/search/controller/SearchFiltering.java

```diff
@@ -489,9 +497,12 @@ private static void setSelectedFilters(Map<String, SearchFilter> filtersByField,
         }
     }
 
-    private static boolean isNotLoggedIn(VitroRequest vreq) {
+    private static Set<String> getCurrentUserRoles(VitroRequest vreq) {
         UserAccount user = LoginStatusBean.getCurrentUser(vreq);
```

Comment:
https://github.com/litvinovg/Vitro/blob/c0b48511e2b7b4446b90a4096ed19de3bae6d7bd/api/src/main/java/edu/cornell/mannlib/vedit/beans/LoginStatusBean.java#L21C1-L21C31

Shouldn't an immutable object be a final class?

Can't the class be extended on the classpath to mutate the static reference to the stack of the `DUMMY_BEAN`?

<img width="950" alt="image" src="https://github.com/vivo-project/Vitro/assets/144841721/c1c0d514-2874-4144-a115-0b373eec9f1a">

This is a spaghetti bowl of session management with the dyadic method containing the non-injected bean of unknown origin.

Reply:
I don't know. Let's create an issue and discuss it there. 
I think it is out of scope for this PR.

Reply:
Yeah, it is out of scope to some degree. Also, the origins of the beans can be traced back to instantiation. Just not sure of the runtime bytecode modifications between RDF interpolation (not radial basis function interpolation) and Java classes.